### PR TITLE
(DOCSP-33119): C++: Update asymmetric sync for 0.3.0-preview

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-33119

### Staged Changes

- [Configure & Open a Realm](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-33119/sdk/cpp/realm-files/configure-and-open-a-realm/#open-a-synced-realm)
- [Stream Data to Atlas](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-33119/sdk/cpp/sync/stream-data-to-atlas/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/cpp/realm-files/configure-and-open-a-realm.txt [hunk 1]
```diff
@@ -208,6 +208,14 @@ To open a synced realm:
       .. literalinclude:: /examples/generated/cpp/flexible-sync.snippet.beta-flexible-sync-prerequisites.cpp
          :language: cpp
 
+      When you open a synced realm, you don't need to pass the pass the
```
Comment (cbullinger) [inline source/sdk/cpp/realm-files/configure-and-open-a-realm.txt:211]:
```suggestion
      When you open a synced realm, you don't need to pass the 
```

File: source/sdk/cpp/realm-files/configure-and-open-a-realm.txt [hunk 2]
```diff
@@ -208,6 +208,14 @@ To open a synced realm:
       .. literalinclude:: /examples/generated/cpp/flexible-sync.snippet.beta-flexible-sync-prerequisites.cpp
          :language: cpp
 
+      When you open a synced realm, you don't need to pass the pass the 
+      models you want the realm to manage into the template parameter list 
+      of the open function. This is a change from the deprecated Alpha API.
+
+      The only exception to this rule is opening a synced realm for 
+      asymmetric sync. For more information, refer to
```
Comment (cbullinger) [inline source/sdk/cpp/realm-files/configure-and-open-a-realm.txt:216]: " opening a synced realm for asymmetric sync"
I like this phrasing, but I don't think I've seen it used -- are we referring to it this way in other places?

**Review by cbullinger — APPROVED** [review]

File: source/sdk/cpp/realm-files/configure-and-open-a-realm.txt (hunk 1 above)
Comment (dacharyc) [reply source/sdk/cpp/realm-files/configure-and-open-a-realm.txt:211]: Oof, good catch, thank you!

File: source/sdk/cpp/realm-files/configure-and-open-a-realm.txt (hunk 2 above)
Comment (dacharyc) [reply source/sdk/cpp/realm-files/configure-and-open-a-realm.txt:216]: Actually, I don't think so - not anymore, anyway. I think we got rid of the asymmetric sync terminology when product renamed the feature to Data Ingest. I'll be more specific here about syncing `asymmetric_object` types.

**Pull request merged**